e2e/test/scenarios/dashboard-filters/dashboard-filters-auto-apply.cy.spec.js

```diff
@@ -507,6 +451,102 @@ describe("scenarios > dashboards > filters > auto apply", () => {
+          .button("Apply")
+          .should("be.visible")
+          .click();
+        getDashboardCard().findByText("Rows 1-5 of 54").should("be.visible");
+      });
+
+      it("should display a toast when a dashboard takes longer than 15s to load", () => {
+        createDashboard();
+        // Not sure why I need to pass a date in this case, but it doesn't work without it.
+        cy.clock(Date.now());
+        openSlowFullAppEmbeddingDashboard({ [FILTER.slug]: "Gadget" });
+        cy.tick(TOAST_TIMEOUT);
+        cy.wait("@cardQuery");
+        undoToast().within(() => {
+          cy.findByText(TOAST_MESSAGE).should("be.visible");
+          cy.button("Turn off").click();
+          cy.wait("@updateDashboard");
+        });
+
+        // In embedding we'll load bookmark after the dashboard is loaded, it's the opposite in normal app because bookmark is cached from somewhere else.
+        // And somehow, dashboard card query will be completed before the dashboard even start to load, and in entity loader it uses `setTimeout`,
+        // so to make sure callback in `setTimeout` is called, we need to advance the clock using cy.tick().
+        cy.tick();
+
+        toggleDashboardInfoSidebar();
+        rightSidebar()
+          .findByLabelText("Auto-apply filters")
+          .should("not.be.checked");
+        filterWidget().findByText("Gadget").should("be.visible");
+
+        // Card height isn't updated because we're mocking clock, that's it's only showing 1 result per page.
+        getDashboardCard().findByText("Rows 1-1 of 53").should("be.visible");
+
+        // card result should be updated after manually updating the filter
+        filterWidget().icon("close").click();
+        dashboardParametersContainer()
+          .button("Apply")
+          .should("be.visible")
+          .click();
+
+        // Card height isn't updated because we're mocking clock, that's it's only showing 1 result per page.
+        getDashboardCard().findByText("Rows 1-1 of 200").should("be.visible");
+      });
+
+      it("should not display a toast when a dashboard takes longer than 15s to load if users have no write access to a dashboard", () => {
+        createDashboard();
+        cy.signIn("readonly");
+        // Not sure why I need to pass a date in this case, but it doesn't work without it.
+        cy.clock(Date.now());
+        openSlowFullAppEmbeddingDashboard({ [FILTER.slug]: "Gadget" });
+        cy.tick(TOAST_TIMEOUT);
+        cy.wait("@cardQuery");
+        undoToast().should("not.exist");
+
+        // In embedding we'll load bookmark after the dashboard is loaded, it's the opposite in normal app because bookmark is cached from somewhere else.
+        // And somehow, dashboard card query will be completed before the dashboard even start to load, and in entity loader it uses `setTimeout`,
+        // so to make sure callback in `setTimeout` is called, we need to advance the clock using cy.tick().
+        cy.tick();
```

Comment:
Actually, we don't necessarily need to call this line, but without it the dashboard header will be loading forever, so I added this for the sake of completeness.